internal/plugin/connectors/http/generic/config.go

```diff
@@ -85,23 +87,45 @@ func newConfig(cfgYAML *ConfigYAML) (*config, error) {
 		cfg.CredentialPatterns[cred] = re
 	}
 
-	// Validate and save header template strings
-	for header, tmplStr := range cfgYAML.Headers {
-		tmpl := newHeaderTemplate(header)
+	cfg.Headers, errs = validateAndSaveTemplateString(cfgYAML.Headers, errs)
+	cfg.QueryParams, errs = validateAndSaveTemplateString(cfgYAML.QueryParams, errs)
+
+	if err := errs.Filter(); err != nil {
+		return nil, err
+	}
+
+	return cfg, nil
+}
+
+func validateAndSaveTemplateString(templates map[string]string,
```

Comment:
I think something shorter may be needed. How about `unmarshalYAMLParams`? Something else?

Reply:
I'm not sure that's accurate, as we've already performed our unmarshalling at this point. Here we're converting the strings to templates, so I think something like "stringToTemplate" would be best.